libraries/AP_Scripting/lua_bindings.cpp

```diff
@@ -250,11 +250,196 @@ const luaL_Reg AP_Logger_functions[] = {
     {NULL, NULL}
 };
 
+
+static int add_param(lua_State *L) {
+    check_arguments(L, 3, "param.add");
+
+    if (lua_type(L, 3) != LUA_TTABLE) {
+        return luaL_error(L, "expected param table for argument 3");
+    }
+
+    size_t num_params = lua_rawlen(L,3);
+    if (num_params == 0) {
+        return luaL_error(L, "need at least one param");
+    }
+
+    uint16_t index = static_cast<uint16_t>(luaL_checkinteger(L, 1));
+
+    // this magic value is to be compared against a automaticaly loaded param at the start of the new table,
+    // the the magics don't match we should reset the new table to defaults 
+    //const uint16_t magic = static_cast<uint16_t>(luaL_checkinteger(L, 2));
+
+    AP_Param::Info *info = (AP_Param::Info *)hal.util->malloc_type(sizeof(AP_Param::Info)*(num_params+1), AP_HAL::Util::Memory_Type::MEM_FAST);
+    struct AP_Param::var_table *table = new AP_Param::var_table;
+
+    for (uint8_t i=0; i<num_params; i++) {
+
+        lua_pushinteger(L, i+1); // lua is 1 indexed
+        lua_gettable(L,-2);
+        if (lua_type(L, -1) != LUA_TTABLE) {
+            gcs().send_text(MAV_SEVERITY_ERROR,"Lua: expected table in param table index %u", i+1);
+            lua_pop (L, 1); // this pops the outer table index
```

Comment:
I'm not sure there is even any point in pooing this if were going to error out of the function anyway?

Reply:
Nope.